src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.HammingDistance.cs

```diff
@@ -0,0 +1,191 @@
+
+                            count += BitOperations.PopCount((~Vector256.Equals(xVec, yVec)).ExtractMostSignificantBits());
+
+                            i += Vector256<T>.Count;
+                        }
+                        while (i <= oneVectorFromEnd);
+
+                        // Process the last vector in the span, masking off elements already processed.
+                        if (i != x.Length)
+                        {
+                            Vector256<T> xVec = Vector256.LoadUnsafe(ref xRef, (uint)(x.Length - Vector256<T>.Count));
+                            Vector256<T> yVec = Vector256.LoadUnsafe(ref yRef, (uint)(x.Length - Vector256<T>.Count));
+
+                            Vector256<T> remainderMask = CreateRemainderMaskVector256<T>(x.Length - i);
+                            xVec &= remainderMask;
+                            yVec &= remainderMask;
+
+                            count += BitOperations.PopCount((~Vector256.Equals(xVec, yVec)).ExtractMostSignificantBits());
+                        }
+                    }
+                }
+                else
+                {
+                    ref T xRef = ref MemoryMarshal.GetReference(x);
+                    ref T yRef = ref MemoryMarshal.GetReference(y);
+
+                    // Process vectors, summing their dot products and squares, as long as there's a vector's worth remaining.
+                    int oneVectorFromEnd = x.Length - Vector128<T>.Count;
+                    int i = 0;
+                    do
+                    {
+                        Vector128<T> xVec = Vector128.LoadUnsafe(ref xRef, (uint)i);
+                        Vector128<T> yVec = Vector128.LoadUnsafe(ref yRef, (uint)i);
+
+                        count += BitOperations.PopCount((~Vector128.Equals(xVec, yVec)).ExtractMostSignificantBits());
+
+                        i += Vector128<T>.Count;
+                    }
+                    while (i <= oneVectorFromEnd);
+
+                    // Process the last vector in the span, masking off elements already processed.
+                    if (i != x.Length)
+                    {
+                        Vector128<T> xVec = Vector128.LoadUnsafe(ref xRef, (uint)(x.Length - Vector128<T>.Count));
+                        Vector128<T> yVec = Vector128.LoadUnsafe(ref yRef, (uint)(x.Length - Vector128<T>.Count));
+
+                        Vector128<T> remainderMask = CreateRemainderMaskVector128<T>(x.Length - i);
+                        xVec &= remainderMask;
+                        yVec &= remainderMask;
+
+                        count += BitOperations.PopCount((~Vector128.Equals(xVec, yVec)).ExtractMostSignificantBits());
+                    }
+                }
+            }
+            else
+            {
+                for (int i = 0; i < x.Length; i++)
+                {
+                    if (!EqualityComparer<T>.Default.Equals(x[i], y[i]))
```

Comment:
Do we want to specialize for the case where `T` is a reference type? (i.e. load the comparer into a local?)

Reply:
Sure. Updated.